src/gui/editors/PianoRoll.cpp

```diff
@@ -4086,6 +4140,42 @@ void PianoRoll::pasteNotes()
 }
 
 
+void PianoRoll::pasteGhostNotes()
+{
+	if(!hasValidPattern()) { return; }
+
+	QString value = Clipboard::getString(Clipboard::MimeType::Default);
+
+	if(!value.isEmpty())
```

Comment:
I'd rather see the inverse of this. Put the failure state inside the `if` instead of the success state.